File: Web/frontend/src/app/models/usuario.ts
export class Usuario {
  id: number;
  id_rol: number;
  nombre: string;
  apellido: string;
  email: string;
  fec_nac: string;
  dni: number;
  tel: string;

  constructor(data: any) {
    this.id = data.id || 0; 
    this.id_rol = data.id_rol || 0;
    this.nombre = data.nombre || '';
    this.apellido = data.apellido || '';
    this.email = data.email || '';
    this.fec_nac = data.fec_nac || '';
    this.dni = data.dni || 0;
    this.tel = data.tel || '';
  }
}

export class UsuarioDTO {
  nombre: string = "";
  apellido: string = "";
  email: string = "";
  password: string = "";
  tel: string = "";
  dni: string = "";
  fec_nac: string = "";
}

export class UsuarioLoginDTO{
  email: string | null | undefined = "";
  password: string | null | undefined = "";
}

File: Movil/app/src/main/java/com/cdp/misrutinas/entidades/Usuario.java
package com.cdp.misrutinas.entidades;

public class Usuario {
    //"CREATE TABLE Usuario (id_usuario INTEGER NOT NULL PRIMARY KEY AUTOINCREMENT, username VARCHAR(20) UNIQUE , apellido VARCHAR(45), nombre VARCHAR(45), dni INTEGER,  email VARCHAR(75) NOT NULL,tel INTEGER, pass VARCHAR(16), active BOOLEAN, id_rol INTEGER, FOREIGN KEY (id_rol) REFERENCES Rol(id_rol))";
    private int idUsuario;
    private String username;
    private String password;
    private String nombre;
    private String apellido;
    private String dni;
    private String email;
    private int id_rol;

    public int getId() {
        return idUsuario;
    }

    public void setId(int idUsuario) {
        this.idUsuario = idUsuario;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellido() {
        return apellido;
    }

    public void setApellido(String apellido) {
        this.apellido = apellido;
    }

    public String getNombreApellido() {
        return nombre + " " + apellido;
    }

    public String getDni() {
        return dni;
    }

    public void setDni(String dni) {
        this.dni = dni;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getUsername() {
        return username;
    }

    public void setUsername(String username) {
        this.username = username;
    }

    public String getPassword() {
        return password;
    }

    public void setPassword(String password) {
        this.password = password;
    }

    public int getId_rol() {
        return id_rol;
    }

    public void setId_rol(int id_rol) {
        this.id_rol = id_rol;
    }
}
